File: RemixedDungeon/src/main/java/com/watabou/pixeldungeon/actors/buffs/MindVision.java

package com.watabou.pixeldungeon.actors.buffs;

import com.nyrds.pixeldungeon.ml.R;
import com.nyrds.platform.util.StringsManager;
import com.watabou.pixeldungeon.Dungeon;
import com.watabou.pixeldungeon.sprites.CharSprite;
import com.watabou.pixeldungeon.ui.BuffIndicator;
import com.watabou.pixeldungeon.utils.GLog;

public class MindVision extends FlavourBuff {

	public static final float DURATION = 20f;


	static public void reportMindVisionEffect() {
		Dungeon.observe();
		if (Dungeon.level.mobs.size() > 0) {
            GLog.i(StringsManager.getVar(R.string.PotionOfMindVision_Apply1));
		} else {
            GLog.i(StringsManager.getVar(R.string.PotionOfMindVision_Apply2));
		}
	}

	@Override
	public int icon() {
		return BuffIndicator.MIND_VISION;
	}

	@Override
	public void detach() {
		super.detach();
		Dungeon.observe();
	}

	@Override
	public void attachVisual() {
        target.showStatus(CharSprite.POSITIVE, StringsManager.getVar(R.string.Char_StaMind));
        target.showStatus(CharSprite.POSITIVE, StringsManager.getVar(R.string.Char_StaVision));
	}
}

File: scripts/buffs/MindVision.lua
---
--- MindVision: see through walls; view recalculated on detach
--- (was actors/buffs/MindVision.java)
---
local RPD  = require "scripts/lib/commonClasses"
local buff = require "scripts/lib/buff"

local CharSprite = luajava.bindClass("com.watabou.pixeldungeon.sprites.CharSprite")

return buff.init{
    desc  = function ()
        return {
            icon          = 0, -- BuffIndicator.MIND_VISION
            name          = "MindVisionBuff_Name",
            info          = "MindVisionBuff_Info",
        }
    end,

    detach = function(self, buff)
        buff.target:observe()
    end,

    attachVisual = function(self, buff)
        buff.target:showStatus(CharSprite.POSITIVE, RPD.textById("Char_StaMind"))
        buff.target:showStatus(CharSprite.POSITIVE, RPD.textById("Char_StaVision"))
    end
}
